lib/github-pages-health-check.rb

```diff
@@ -65,11 +66,15 @@ def valid_domain?
       PublicSuffix.valid? domain
     end
 
-    # Is this domain an SLD, meaning a CNAME would be innapropriate
+    # Is this domain an apex domain, meaning a CNAME would be innapropriate
     def apex_domain?
-      PublicSuffix.parse(domain).trd == nil
-    rescue
-      false
+      return @apex_domain if defined?(@apex_domain)
+
+      answers = Resolv::DNS.open { |dns|
+        dns.getresources(absolute_domain, Resolv::DNS::Resource::IN::NS)
```

Comment:
Makes sense. I think Net::DNS just wraps resolve, and it's a core lib, so it's not like we're adding an external dependency. If resolve works more cleanly, lets use that. :+1:.